utils/bids/download_run_level.py

```diff
@@ -0,0 +1,364 @@
+        src_data (boolean): download source data (dicoms) as well
+        folders (list): only include the listed folders, if empty include all
+        dry_run (boolean): don't actually download data if True
+        do_validate_bids (boolean): run bids-validator after downloading bids data
+
+    Returns:
+        err_code (int): tells a bit about the error:
+            0    - no error
+            1..9 - error code returned by bids validator
+            10   - BIDS validation errors were detected
+            11   - the validator could not be run
+            12   - TypeError while analyzing validator output
+            20   - running at wrong level
+            21   - BIDSExportError
+            22   - validator exception
+            23   - attempt to download unknown acquisition
+            24   - destination does not exist
+            25   - download_bids_dir() ApiException
+            26   - no BIDS data was downloaded
+
+    Note: information on BIDS "folders" (used to limit what is downloaded)
+    can be found at https://bids-specification.readthedocs.io/en/stable/99-appendices/04-entity-table.html.
+    """
+
+    extra_tree_text = ""  # Text to be added to the end of the tree HTML file
+
+    run_level = hierarchy["run_level"]
+
+    # Show the complete destination hierarchy in the tree html output for
+    # clarity
+    extra_tree_text += f"run_level is {run_level}\n"
+    for key, val in hierarchy.items():
+        extra_tree_text += f"  {key:<18}: {val}\n"
+    extra_tree_text += f'  {"folders":<18}: {folders}\n'
+    if src_data:
+        extra_tree_text += f'  {"source data?":<18}: downloaded\n'
+    else:
+        extra_tree_text += f'  {"source data?":<18}: not downloaded\n'
+    if dry_run:
+        extra_tree_text += f'  {"dry run?":<18}: Yes\n'
+    else:
+        extra_tree_text += f'  {"dry run?":<18}: No\n'
+    extra_tree_text += "\n"
+
+    err_code = 0  # assume no error
+
+    if run_level == "no_destination":
+        msg = "Destination does not exist."
+        log.critical(msg)
+        extra_tree_text += f"ERROR: {msg}\n"
+        bids_path = None
+        err_code = 24  # destination does not exist
+
+    else:
+
+        if gtk_context.destination["type"] == "analysis":
+            pass
+
+        elif gtk_context.destination["type"] == "acquisition":
+            log.info("Destination is acquisition, changing run_level to " "acquisition")
+            acquisition = gtk_context.client.get_acquisition(
+                gtk_context.destination["id"]
+            )
+            hierarchy["acquisition_label"] = acquisition.label
+            extra_tree_text += (
+                f'  {"acquisition_label":<18}: changed to ' + f"{acquisition.label}\n\n"
+            )
+            run_level = "acquisition"
+
+        else:
+            log.info(
+                'The destination "%s" is not an analysis or acquisition.',
+                gtk_context.destination["type"],
```

Comment:
`get_analysis_run_level_and_hierarchy()` does not necessarily have to be called before downloading the data here.  This may be called in unforeseen ways.